### ISSUE TITLE
Support Azure App Service Certificate Orders

### ISSUE THREAD

**Issue by Moeser** [terraform-providers/terraform-provider-azurerm#2038]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Azure has an api call to order app certs that are signed.  They can also be added to the key vault.  Note that this is NOT the same as the azurerm_key_vault_certificate resource.  Docs from Microsoft in references section.

This call is more similar to the AWS call to create a signed ACM cert `aws_acm_certificate`, and doesn't require setting up a certificate authority account.  It does cost money though, vs ACM, so be careful when testing.

### New or Affected Resource(s)

* azurerm_app_service_certificate_order ?

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
resource "azurerm_app_service_certificate_order" "example" {
  name = "example-cert-order"
  resource_group_name = "example-rg"
  certificate_product_type = "StandardDomainValidatedWildCardSsl"
  auto_renew = true
  distinguished_name = "CN=*.example.some-example-domain.com"
  location = "${var.azure_location}"
  validity_in_years = 1
  tags = {
    tagname = "example-tag"
  }
}

output "status" {
  value = "${azurerm_app_service_certificate_order.example.status}"
}

output "cert" {
  value = "${azurerm_app_service_certificate_order.example.signed_certificate}"
}

...

```

### References

These links look similar, but they're different documents.

* https://docs.microsoft.com/en-us/rest/api/appservice/appservicecertificateorders/createorupdatecertificate

* https://docs.microsoft.com/en-us/rest/api/appservice/appservicecertificateorders/createorupdate

**Comment by LaurentLesle** [comment]
GO SDK to support this request: https://github.com/Azure/azure-sdk-for-go/blob/master/services/web/mgmt/2018-02-01/web/appservicecertificateorders.go

**Comment by tombuildsstuff** [comment]
Fixed via #4454 - thanks @hund030
